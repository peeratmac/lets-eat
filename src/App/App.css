@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  /* background-image: url('../assets/cafe-img.jpeg'); */
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

button {
  display: inline-block;
  background: #4e6d9c;
  color: white;
  padding: 0.4rem 1.3rem;
  font-size: 1rem;
  border: none;
  border-radius: 20px;
  cursor: pointer;
  margin-right: 0.5rem;
  transition: opacity 0.2s ease-in;
  outline: none;
}

input {
  border-radius: 10px;
  display: block;
  width: 100%;
  padding: 0.4rem;
  font-size: 1.2rem;
  border: 1px solid #ccc;
  margin: 15px;
}

.make-reservation-button {
  margin: 30px;
  background: rgb(87, 159, 187);
}
