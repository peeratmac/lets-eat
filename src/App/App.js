import React, { Component } from 'react';
import { getAllReservations } from '../apiCalls';
import './App.css';
class App extends Component {
  constructor() {
    super();
    this.state = {
      allReservationsData: [],
      error: ''
    };
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(data => this.setState({ allReservationsData: data }))
      .catch(error => this.setState({ error: error.message }));
  }

  render() {
    return (
      <div className='App'>
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'></div>
        <div className='resy-container'></div>
      </div>
    );
  }
}

export default App;
