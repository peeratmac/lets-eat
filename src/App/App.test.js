import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import ReactDOM from 'react-dom';

describe('APP', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the App Component Snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
