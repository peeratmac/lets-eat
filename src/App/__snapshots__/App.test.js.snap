// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the App Component Snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <FormInput
      makeReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationContainer
      cancelReservation={[Function]}
      reservations={Array []}
    />
  </div>
</div>
`;
